script.js

```diff
@@ -0,0 +1,958 @@
+*/
+const dealButtonClickEvent = () => {
+  if (canClick === true) {
+    pointsCredited.innerHTML = '';
+    winMsgBanner.style.visibility = 'hidden';
+    loseMsgBanner.style.visibility = 'hidden';
+    // check playerBet in case bet amount entered manually
+    playerBet = Number(betAmount.value);
+    highlightColumn();
+    if (playerBet > 0 && playerBet < 6 && playerPoints > 0) {
+      if (gameMode === 'default') {
+        cardNameTally = {};
+        suitTally = {};
+        straightTally = {};
+        checkDeck();
+        for (let i = 0; i < 5; i += 1) {
+          const card = deck.pop();
+          hand.push(card);
+        }
+        // hand = hardCodeHand;
+        dealSound.play();
+        setTimeout(displayCards, 500);
+        board.appendChild(buttonRow);
+        gameMode = 'secondDeal';
+        setTimeout(() => {
+          gameText.innerHTML = 'Select which cards to keep (if any) and click [ DEAL ]<br>Highlighted cards will remain in your hand'; }, 1000);
+      } else if (gameMode === 'secondDeal') {
+        canClick = false;
+        dealSound.play();
+        gameText.innerHTML = 'analyzing results ...';
+
+        setTimeout(() => {
+          reDeal();
+          gameMode = 'default';
+        }, 600);
+
+        setTimeout(() => {
+          calcHandScore();
+          hand = [];
+          betAmount.value = 0;
+          playerBet = Number(0);
+          const rowID = `#${winID}`;
+          if (score > 0) {
+            document.querySelectorAll(`${rowID}`).forEach((el) => {
+              el.classList.add('highlight-column');
+            });
+          } }, 2000);
+
+        setTimeout(() => {
+          if (score === 5) {
+            pointsCredited.style.visibility = 'hidden';
+          } else {
+            creditNumber.innerHTML = `${playerPoints}`;
+            pointsCreditSound.play();
+            creditNumber.classList.add('flash');
+            pointsCredited.style.visibility = 'visible';
+          }
+        }, 5000);
+
+        setTimeout(() => {
+          gameText.innerHTML = 'Enter bet and click [ DEAL ] to play another round';
+          boardCard.innerHTML = '';
+          winMsgBanner.style.visibility = 'hidden';
+          loseMsgBanner.style.visibility = 'hidden';
+          initDeck();
+          document.querySelectorAll('[id^="bet"]').forEach((el) => {
+            el.classList.remove('highlight-column');
+          });
+          document.querySelectorAll('[id^="win"]').forEach((el) => {
+            el.classList.remove('highlight-column');
+          });
+          canClick = true;
+          score = 0;
+          flipCardsSound.play();
+          pointsCredited.style.visibility = 'hidden';
+        }, 8000);
+      }
+      else if (playerPoints <= 0) {
+        output('You have no more credits left<br>Buy more credits to continue playing');
+      } }
+    else { gameText.innerHTML = 'Use [ MAX ] [ + ] [ - ] buttons<br>to enter your bet and then click [ DEAL ]'; }
+  }
+};
+
```

Comment:
consider breaking the content of this function up into separate helper functions so that your code is easier to read